workers/handshake-manager/src/manager/matching/internal_engine.rs

```diff
@@ -197,30 +156,12 @@ impl HandshakeExecutor {
         rx.await
             .map_err(err_str!(HandshakeManagerError::TaskError))? // RecvError
             .map_err(err_str!(HandshakeManagerError::TaskError)) // TaskDriverError
-            .map(|_| true)
     }
 
     // -----------
     // | Helpers |
     // -----------
 
-    /// Check if two wallets may match
-    fn wallets_can_match(&self, w1: WalletIdentifier, w2: WalletIdentifier) -> bool {
-        // Same wallet
-        if w1 == w2 {
```

Comment:
We still need to do this check somewhere, and we no longer do. I think the best place would be in `get_match_candidates`, though it would mean two extra state lookups for wallet IDs there. conversely, it could be done in `try_settle_match`, where we already look up the wallet IDs